---
title: [deepTools](http://deeptools.ie-freiburg.mpg.de/)
---
<div class='center'>
<a href='http://deeptools.ie-freiburg.mpg.de/'><img src="/src/use/deeptools/deeptools.png" alt="deepTools" height="200" /></a>
</div>





<div class='deploymentbox'>
 Deployment:: **[deepTools](/src/community/deployment/deep-tools/index.md)**
 Domain:: Tools for the visualization, quality control and normalization of data from high-throughput DNA sequencing experiments.
 Owners:: [Bioinformatics](http://www.ie-freiburg.mpg.de/bioinformaticsfac) and [Deep-Sequencing](http://www.ie-freiburg.mpg.de/deepfac) Units at the [Max Planck Institute for Immunobiology and Epigenetics](http://www3.ie-freiburg.mpg.de/).
 Audience:: [Publicly accessible](/src/use/index.md). 
 User Base:: Unknown
 Server Topology:: Standalone server
 Compute:: 16 processors 
 Memory:: 141GB
 Storage:: Directly attached
 Disk Size:: 2TB
 User Management:: Account creation is open
</div>

deepTools offers multiple methods for highly-customizable data visualization that immensely aid hypothesis generation and data interpretation. It also offers all the tools needed to create coverage files in standard bedGraph and bigWig file formats allowing various normalization procedures and comparisons between two files (for example, treatment and control).

## Domain

## Community

## Compute Infrastructure

See [this NAR supplement](http://nar.oxfordjournals.org/content/suppl/2014/05/05/gku365.DC1/nar-00319-web-b-2014-File009.pdf).

<table>
  <tr>
    <th colspan=2> Hardware </th>
  </tr>
  <tr>
    <td> processors </td>
    <td> 16 </td>
  </tr>
  <tr>
    <td> RAM </td>
    <td> 141 GB </td>
  </tr>
  <tr>
    <td> storage (directly attached) </td>
    <td> 2 TB </td>
  </tr>
  <tr>
    <td colspan=2 style=" border: none;"> </td>
  </tr>
  <tr>
    <th colspan=2> Galaxy Performance Settings </th>
  </tr>
  <tr>
    <td> concurrent users </td>
    <td> unlimited </td>
  </tr>
  <tr>
    <td> CPU cores per job </td>
    <td> 6 </td>
  </tr>
  <tr>
    <td> concurrent jobs per user </td>
    <td> 1 </td>
  </tr>
  <tr>
    <td> concurrent jobs </td>
    <td> 2 </td>
  </tr>
  <tr>
    <td> queueing system </td>
    <td> Galaxy built-in </td>
  </tr>
  <tr>
    <td colspan=2 style=" border: none;"> </td>
  </tr>
  <tr>
    <th colspan=2> Exemplary run time </th>
  </tr>
  <tr>
    <td> BAM file </td>
    <td> 170 M reads </td>
  </tr>
  <tr>
    <td> BED file </td>
    <td> 10000 regions </td>
  </tr>
  <tr>
    <td> bamCoverage </td>
    <td> 10 minutes </td>
  </tr>
  <tr>
    <td> computeMatrix </td>
    <td> 1 minutes </td>
  </tr>
</table>


## Links

* [deepTools entry](/src/use/deeptools/index.md) in the [Public Galaxy Server list](/src/use/index.md)
* [deepTools home page at GitHub](http://deeptools.github.io/)
* Fidel Ramírez, Friederike Dündar, Sarah Diehl, Björn A. Grüning, and Thomas Manke. [deepTools: a flexible platform for exploring deep-sequencing data](http://nar.oxfordjournals.org/cgi/content/abstract/gku365) *Nucl. Acids Res.* first published online May 5, 2014 doi:10.1093/nar/gku365
* *[deepTools: a flexible platform for exploring deep-sequencing data](/src/events/gcc2014/abstracts/talks/index.md#deeptools-a-flexible-platform-for-exploring-deep-sequencing-data)* presentation by [Sarah Diehl](http://www.ie-freiburg.mpg.de/1892622/employee_page?c=1896591&employee_id=26993) at [GCC2014](/src/events/gcc2014/index.md)
