---
autotoc: true
title: March 2013 Galaxy Update
---
<div class='right'></div>



<div class='left'><a href='/GalaxyUpdates'><img src='/Images/Logos/GalaxyUpdate200.png' alt='Galaxy Updates' width=150 /></a></div>

Welcome to the March 2013 *[Galaxy Update](/GalaxyUpdates)*, a monthly summary of what is going on in the Galaxy community. *[Galaxy Updates](/GalaxyUpdates)* complement the *[Galaxy Development News Briefs](/src/DevNewsBriefs/index.md)* which accompany new Galaxy releases and focus on Galaxy code updates.

<br />

## New Public Galaxy Servers

<div class='left'><a href='http://www.bioinformatica.ucr.ac.cr:8080/'><img src='/PublicGalaxyServers/UCostaRicaLogo.png' alt='University of Costa Rica' /></a></div> The [University of Costa Rica](http://www.ucr.ac.cr/) School of Medicine, [Master in Bioinformatics and System Biology](http://www2.sep.ucr.ac.cr/MaestriasAcademicas/CBIOMEDICAS/Infodescrip.html) and [BREL (Bioinformatics Research Laboratory)](http://www.ucrbrel.com/) have made the [Galaxy NELLY](http://www.bioinformatica.ucr.ac.cr:8080/) server publicly accessible.  NELLY features the standard Galaxy tool set.  It requires a login, but anyone can create a login.

This server joins the [growing list of public Galaxy servers](/src/PublicGalaxyServers/index.md).

## New Papers

<div class='left'>
<table>
  <tr>
    <th> # </th>
    <th> Tag </th>
  </tr>
  <tr>
    <td style=" text-align: right;"> 15 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/methods'>methods</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 11 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/workbench'>workbench</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/tools'>tools</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/unknown'>unknown</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/isgalaxy'>isgalaxy</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/reproducibility'>reproducibility</a> </td>
  </tr>
</table>

</div>

These papers may be of interest to the Galaxy community. 

* "[RepeatExplorer: a Galaxy-based web server for genome-wide characterization of eukaryotic repetitive elements from next generation sequence reads](http://dx.doi.org/10.1093/bioinformatics/btt054)," by Petr Novák, Pavel Neumann, Jiří Pech, Jaroslav Steinhaisl, Jiří Macas, *Bioinformatics* (1 February 2013)
* "[A unified framework for the deployment, exposure and access of HPC applications as services in clouds](http://dx.doi.org/10.1016/j.future.2013.01.014)," by Adam K. L. Wonga, Andrzej M. Goscinskia, *Future Generation Computer Systems* (February 2013)

There are now over [880 papers in the Galaxy CiteULike Group](http://www.citeulike.org/group/16008/).

<br />

## Who's Hiring

<div class='right'><a href='/GalaxyIsHiring'><img src='/GalaxyIsHiring/GalaxyIsHiringWordCloud2.png' alt='Please Help! Yes you!' width="200" /></a></div>

The Galaxy is expanding! Please help it grow.

* **The [Galaxy Project is hiring software engineers and post-docs](/src/GalaxyIsHiring/index.md)!** at both Emory and Penn State.
* **[Bioinformatics Trainter/Analyst](http://bit.ly/14xgrVK)**, [Morgridge Institute for Research](http://discovery.wisc.edu/morgridge/), Madison, Wisconsin, United States
* Poste [d’ingénieur bio-informaticien en métagénomique](http://bit.ly/VyQvUL) au Laboratoire d’Analyses Bioinformatiques en Génomique et Métabolisme (LABGeM)

Got a Galaxy-related opening? Send it to outreach@galaxyproject.org and we'll put it in the [Galaxy News feed](/News) and include it in next month's [update](/src/GalaxyUpdates/index.md).


## GCC2013

<div class='center'><a href='/Events/GCC2013'><img src='/Images/Logos/GCC2013Logo400.png' alt='2013 Galaxy Community Conference (GCC2013)' width="300" /></a> </div>

### Registration, Talk/Poster Submission Open

**[Early registration](/Events/GCC2013/Register) and [paper and poster abstract submission](/Events/GCC2013/Abstracts) are now open for the [2013 Galaxy Community Conference (GCC2013)](/Events/GCC2013).**  GCC2013 will be held 30 June through July 2 in Oslo Norway, at the [University of Oslo](http://uio.no).

<div class='right'><a href='/Events/GCC2013/TrainingDay'><img src='/Images/Logos/GCC2013TrainingDayLogo200.png' alt='2013 Galaxy Community Conference (GCC2013) Training Day' width="150" /></a></div>

[GCC2013](/Events/GCC2013) is an opportunity to participate in two full days of presentations, discussions, poster sessions, keynotes, lightning talks and breakouts, all about high-throughput biology and the tools that support it. The conference also includes a [Training Day](/src/Events/GCC2013/TrainingDay/index.md) for the second year in a row, this year with more in-depth topic coverage, more concurrent sessions, and more topics.

**[|Register early](/Events/GCC2013/Register)** and *save up to 75% off regular registration costs.*  Early registration is very affordable, with combined registration ([Training Day](/src/Events/GCC2013/TrainingDay/index.md) + main meeting) starting at ~ €95 for post-docs and students.  Registering early also assures you a spot in the Training Day workshops you want to attend.  Once a session becomes full, it will be closed to new registrations.  Early registration closes 24 May.

**[Abstract submission](/src/Events/GCC2013/Abstracts/index.md)** for both oral presentations and posters is also open.  Abstract submission for oral presentations closes 12 April, and poster abstract submission closes 3 May.  Please consider presenting your work. If you are dealing with big biological data, then this meeting wants to hear about your work.

### Sponsorships

<div class='right'><a href='http://lifetech.com/'><img src='/Images/Logos/IonTorrentLogo340.png' alt='Ion Torrent by Life Technologies' width="140" /></a><br /> 
<div class='right'><a href='http://emc.com/'><img src='/Images/Logos/EMCLogo.png' alt='EMC Isilon' width="120" /></a><br /><div class='right'><a href='http://illumina.com/'><img src='/Images/Logos/IlluminaLogo250.png' alt='Illumina' width="100" /></a></div></div></div>

We are pleased to announce several new [sponsorships](/src/Events/GCC2013/Sponsorships/index.md) for GCC2013:

* [Ion Torrent](http://lifetech.com/) will be the exclusive Platinum Sponsor for the event. 
* [EMC Isilon](http://emc.com/) will be a Gold Sponsor for [GCC2013](/src/Events/GCC2013/index.md). 
* [Illumina](http://illumina.com/) will be a Bronze Sponsor for [GCC2013](/src/Events/GCC2013/index.md).

There are still several [sponsorships](/Events/GCC2013/Sponsorships) still available. Please contact the [Organizers](/src/Events/GCC2013/Organizers/index.md) if your organization would like to help sponsor this event. 

## March GalaxyAdmins Web Meetup

<div class='left'><a href='/Community/GalaxyAdmins/Meetups/2013_03_20'><img src='/Images/Logos/GalaxyAdmins.png' alt='March Galaxy Admins Web Meetup' height="50" /></a></div>

The next [GalaxyAdmins Meetup](/Community/GalaxyAdmins/Meetups/2013_03_20) will be held [March 20, 2013](/Community/GalaxyAdmins/Meetups/2013_03_20) and feature [hailiang DOT mei AT nbic DOT nl](Hailiang "Leon" Mei), and [david.van.enckevort@nbic.nl](David van Enckevort) speaking on *[NBIC Galaxy](http://galaxy.nbic.nl/) at [SURFsara's HPC cloud](https://www.surfsara.nl/).* 

[NBIC Galaxy (Andromeda)](http://galaxy.nbic.nl/) was migrated to a HPC cloud hosted by Surfsara in September 2012. In this presentation, we will present the setup of this HPC cloud and the architecture of our NBIC Galaxy. We will also share our experience on the installation of the NBIC Galaxy using the Cloudman scripts. The presentation will finish with a list of issues and our possible future plans.

See the [meetup page](/src/Community/GalaxyAdmins/Meetups/2013_03_20/index.md) for details.

## Galaxy at GMOD 2013 and Biocuration 2013

<div class='right'> <a href='/gmod:April_2013_GMOD_Meeting'><img src='/Images/Logos/GMOD2013Logo240.png' alt='GMOD 2013' height="80" /></a>
<a href='http://www.ebi.ac.uk/biocuration2013/content/workshop-descriptions'><img src='/Images/Logos/ISBLogoWAcronym223.png' alt='Biocuration 2013' height="70" /></a></div>

Galaxy will [have a presence](/News/GalaxyGMODBiocuration2013) at both the [GMOD 2013](http://gmod.org/wiki/April 2013 GMOD Meeting) and [Biocuration 2013](http://www.ebi.ac.uk/biocuration2013/home).  Both are being held in Cambridge, UK in early April.  If you are interested in either community, then it would be great to see you there.

GMOD is also running a [community-wide survey](http://gmod.org/survey.html) through March 1. The survey "aims to find out how you are using GMOD, what you find useful (or otherwise), what support you need, and some information on the GMOD components you use." The survey results will be published and past surveys have been extremely useful to the project.  If that's not enough motivation, you can enter to win a free genome profile from 23andMe.


## Other Upcoming Events and Deadlines

<div class='center'>
<a href='http://conf.abrf.org/'><img src='/Images/Logos/ABRF2013LogoSquare300.png' alt='ABRF 2013' height="80" /></a>
<a href='http://www.qfab.org/events/upcoming/'><img src='/Images/Logos/QFABLogo.png' alt='QFAB Workshops' height="85" /></a>
<a href='http://www.jgi.doe.gov/meetings/usermeeting/workshops.html#rna-tech'><img src='/Images/Logos/JGILogoSmall.jpg' alt='RNA Technologies and Analysis Workshop' height="85" /></a>
</div>

<div class='right'><a href='http://bit.ly/gxycal'><img src='/Images/Icons/CalendarIcon.gif' /></a></div> 
See the [Galaxy Events Google Calendar](http://bit.ly/gxycal) for details on these and other events.

### Events

<table>
  <tr class="th" >
    <th> Date </th>
    <th> Topic/Event </th>
    <th> Venue/Location </th>
    <th> Contact </th>
  </tr>
  <tr>
    <th> March 2-5 </th>
    <td> <em>Accessible, Transparent and Reproducible Analysis With Galaxy</em>, part of <div class='indent'><a href='http://conf.abrf.org/index.cfm/page/satellite_workshops/SW1.htm'>SW1: Application of NGS Platforms for Whole Transcriptome and Genome Analysis</a></div> <em>Galaxy for Core Facilities</em>, part of <div class='indent'>"<a href='http://conf.abrf.org/index.cfm/page/2013ConfProg/Genomics_Track.htm'>W6: Community Resource Solutions to Analyzing Large Genomic Data Sets</a>"</div> </td>
    <td> </strong><a href='http://conf.abrf.org/'>ABRF 2013</a><strong><br />Palm Springs, California, United States </td>
    <td> <a href='/DaveClements'>Dave Clements</a> </td>
  </tr>
  <tr>
    <th> March 8 </th>
    <td> <em><a href='http://bit.ly/UBiHcK'>RNA-Seq Analysis Using Galaxy</a></em> </td>
    <td> <a href='http://www.qfab.org/'>QFAB</a>, Brisbane, Australia </td>
    <td> <a href="mailto:m DOT crowe AT qfab DOT org">Mark Crowe</a> </td>
  </tr>
  <tr>
    <th> March 20 </th>
    <td> <em><a href='/Community/GalaxyAdmins/Meetups/2013_03_20'>NBIC Galaxy at SURFsara's HPC cloud</a></em> </td>
    <td> <a href='/Community/GalaxyAdmins/Meetups/2013_03_20'>GalaxyAdmins web meetup</a> </td>
    <td> <a href="mailto:hailiang DOT mei AT nbic DOT nl">Hailiang "Leon" Mei</a>,<a href="mailto:david.van.enckevort@nbic.nl">David van Enckevort</a> </td>
  </tr>
  <tr>
    <th> March 22 </th>
    <td> <em><a href='http://bit.ly/152uov8'>Introduction to Variant Detection Using Galaxy</a></em> </td>
    <td> <a href='http://www.qfab.org/'>QFAB</a>, Brisbane, Australia </td>
    <td> <a href="mailto:m DOT crowe AT qfab DOT org">Mark Crowe</a> </td>
  </tr>
  <tr>
    <th> March 26-28 </th>
    <td> <em><a href='http://www.jgi.doe.gov/meetings/usermeeting/workshops.html#rna-tech'>RNA Technologies and Analysis Workshop</a></em> </td>
    <td> <a href='http://www.jgi.doe.gov/meetings/usermeeting/'>DOE JGI User Meeting</a> </td>
    <td> Zhong Wang, <a href='http://www.jgi.doe.gov/whoweare/pennacchio.html'>Len Pennacchio</a> </td>
  </tr>
  <tr>
    <th> April 5-6 </th>
    <td> <em><a href='http://gmod.org/wiki/April 2013 GMOD Meeting'>2013 GMOD Meeting</a></em> </td>
    <td> Cambridge, United Kingdom, immediately prior to <a href='http://www.ebi.ac.uk/biocuration2013/'>Biocuration 2013</a> </td>
    <td> <a href='/DaveClements'>Dave Clements</a> </td>
  </tr>
  <tr>
    <th> April 7-10 </th>
    <td> <em><a href='http://www.ebi.ac.uk/biocuration2013/content/workshop-descriptions'>GO Galaxy Workshop</a></em> </td>
    <td> <a href='http://www.ebi.ac.uk/biocuration2013/home'>Biocuration 2013</a>, Cambridge, United Kingdom </td>
    <td> <a href='/DaveClements'>Dave Clements</a>, <a href='http://www.berkeleybop.org/person/suzanna-lewis'>Suzanna Lewis</a> </td>
  </tr>
  <tr>
    <th> May 14-16 </th>
    <td> Tutorial: <em>Exploring and Enabling Biomedical Data Analysis with Galaxy</em> </td>
    <td> <a href='http://www.iscb.org/glbio2013'>Great Lakes Bioinformatics Conference (GLBIO) 2013</a>, Pittsburgh, Pennsylvania, United States </td>
    <td> <a href='/anton'>Anton Nekrutenko</a> </td>
  </tr>
  <tr>
    <th> June 6-7 </th>
    <td> <em><a href='http://bit.ly/VyBEcS'>Informatics on High Throughput Sequencing Data Workshop</a></em> </td>
    <td> Toronto, Ontario, Cananda </td>
    <td> <a href='http://bioinformatics.ca/person/cbw-experts/francis-ouellette'>Francis Ouellette </a> </td>
  </tr>
  <tr>
    <th> June 8-11 </th>
    <td> <em><a href='https://www.eshg.org/sunday2013.0.html#W1'>WS08: Biomedical Data Analysis with Galaxy</a></em> </td>
    <td> Workshop at <a href='https://www.eshg.org/eshg2013.0.html'>European Human Genetics Conference (ESHG2013)</a>, Paris, France </td>
    <td> <a href='/anton'>Anton Nekrutenko</a>, <a href='/EnisAfgan'>Enis Afgan</a> </td>
  </tr>
  <tr>
    <th> June 30 - July&nbsp;2 </th>
    <td> </strong><em><a href='/Events/GCC2013'>2013 Galaxy Community Conference (GCC2013)</a><strong></em> </td>
    <td> University of Oslo, Oslo, Norway </td>
    <td> <a href='/Events/GCC2013/Organizers'>Organizers</a> </td>
  </tr>
</table>



## Galaxy Distributions

**[February 8, 2013 Galaxy Distribution](/src/DevNewsBriefs/2013_02_08/index.md)**

<div class='indent'>
<div class='right'><a href='/Tool Shed'><img src='/Images/NewsGraphics/2013_02_6-confirm-dependency-installation-sm2.png' alt='' width="220px" /></a></div>
**Highlights:**
* *Improvements* to our [release process](/DevNewsBriefs/2013_02_08#improvements_to_release_process). **Release tag must be used in the hg update command to upgrade**. More at **[getgalaxy.org](/src/Admin/GetGalaxy/index.md)**.
* Tool Shed *[Complex repository dependencies](/DefiningRepositoryDependencies#complex_repository_dependencies_tool_dependency_definitions_that_contain_repository_dependency_definitions)* are introduced, streamlining core dependency use across individual tools. 
* Also updated in the Tool Shed: multiple repository installation, dependency installation (when defined - see graphic on right for an example), and many usability enhancements and fixes.
* New *[Bedgraph-to-bigwig](/Learn/Datatypes#bedgraph)* tool plus *Filter* tool updated.
* *Workflows* now include option to export an image and the core *Framework* now allows more unified reference genome usage and access.
* *Ten Community Pull-Requests* incorporated, plus another contribution to the tool Shed, addressing general usability, API, tools, error handling, workflows, and more. Thanks!!
* Plus bug fixes and related enhancements for visualizations, histories, workflows, and tools.

**[http://getgalaxy.org](http://getgalaxy.org)**

**[http://bitbucket.org/galaxy/galaxy-dist](http://bitbucket.org/galaxy/galaxy-dist)**

**[http://galaxy-dist.readthedocs.org](http://galaxy-dist.readthedocs.org)**

```
new: $ hg clone https://bitbucket.org/galaxy/galaxy-dist#stable galaxy-dist

upgrade: $ hg pull 
$ hg update release_2013.02.08
```

</div>

## Tool Shed Contributions

<div class='right'><a href='http://toolshed.g2.bx.psu.edu/'><img src='/Images/Logos/ToolShed.jpg' alt='Galaxy Tool Shed' width=150 /></a></div>

* viralcloud: Viral Cloud Resource (VCR) includes viral assembly & VIGOR
* cummeRbund: !CuffDiff output analysis
* fastq_sync: Check or repair synchronization of a pair of fastq datasets
* velvet_optimiser: Updated version of velvet optimiser wrapper with all features
* tophat_stats_pe: Calculate mapping statistics from Tophat BAM files
* [blastxml_to_top_descr](http://bit.ly/14NU5zp): Make table of top BLAST match descriptions 
* [qualimap_suite](http://bit.ly/12Lq2FK): [detect biases in sequencing and/or mapping in SAM/BAM data](http://qualimap.bioinfo.cipf.es)
* [smalt](http://bit.ly/UnzH6j): efficiently align DNA sequencing reads with a reference genome 
* [rdiff](http://bit.ly/Xh4Iss): detection of differential RNA processing from RNA Sequncing data 
* [pileup_to_vcf](http://bit.ly/YeaafR): convert a samtools mpileup pileup to vcf 
* [predictnls](http://bit.ly/XCyM0y/) : Python reimplementation of predictNLS for Galaxy 

## Other News

* A [Software Carpentry bootcamp](http://bit.ly/14EmcSA) has been scheduled in Oslo immediately after [GCC2013](/src/Events/GCC2013/index.md).
* An improved [Galaxy Project issues board](http://bit.ly/gxytrello) is out
* Going to [GCC2013](/Events/GCC2013)?  [SocBiN / PTBI](http://bit.ly/SocBiN2013) is right before; Paper and early registration deadline is 15 March 
* **[Introduction to Galaxy 2013 Bootcamp](http://training.bioinformatics.ucdavis.edu/docs/2013/02/bootcamp/galaxy/index.html)** by Nikhil Joshi, Bioinformatics Core, UC Davis Genome Center
* **[Galaxy Tutorials](https://genome.edu.au/wiki/Galaxy_Tutorials)** Including *RNA-Seq, Variant Detection,* and *Genome Assembly* from The [Genomics Virtual Lab](https://genome.edu.au/)
* [CloudBioLinux can now install genome indices from the Galaxy rsync server](https://github.com/chapmanb/cloudbiolinux/commit/208a59bb9ec19b1b633b0319259e313d4fae7952). 
* [Uploading NGS data to Amazon S3 for use in Galaxy](http://bit.ly/137rZfM)
