---
---
* [Wiki Root](/src/toolshed/index.md)
* [All monthly summaries](/src/toolshed/contributions/index.md)

# Tools

* *From [gdroc](https://toolshed.g2.bx.psu.edu/view/gdroc):*
  * [rave](https://toolshed.g2.bx.psu.edu/view/gdroc/rave):  RAVE (Rapid Allelic Variant Extractor) : Extract data from binary PLINK file

* *From [iuc](https://toolshed.g2.bx.psu.edu/view/iuc):*
  * [featurecounts](https://toolshed.g2.bx.psu.edu/view/iuc/featurecounts): Counts reads aligned to annotated genes in a reference genome from SAM or BAM files.
    * [quast](https://toolshed.g2.bx.psu.edu/view/iuc/quast): Quast (Quality ASsessment Tool) evaluates genome assemblies by computing various metrics, including - N50, length for which the collection of all contigs of that length or longer covers at least 50% of assembly length, - NG50, where length of the reference genome is being covered, - NA50 and NGA50, where aligned blocks instead of contigs are taken, - misassemblies, misassembled and unaligned contigs or contigs bases, - genes and operons covered.
    * [scikit_bio_diversity_beta_diversity](https://toolshed.g2.bx.psu.edu/view/iuc/scikit_bio_diversity_beta_diversity): Wrapper for the scikit-bio tool suite: Beta Diversity scikit-bio is an open-source, BSD-licensed, python package providing data structures, algorithms, and educational resources for bioinformatics
    * [data_source_iris_tcga](https://toolshed.g2.bx.psu.edu/view/iuc/data_source_iris_tcga): Data source tool Retrieve data from Information Retrieval and Integration System for cancer genomic data from The Cancer Genome Atlas (IRIS-TCGA). IRIS-TCGA is an online web service for searching, retrieving and integrating genomic data: http://bioinf.iasi.cnr.it/iristcga/index.php.
    * [bwameth](https://toolshed.g2.bx.psu.edu/view/iuc/bwameth): Fast and accurate alignment of BS-seq reads. Supports single-end and paired-end alignments and gapped alignments and is faster and more sensitive than many other tools
    * [star_fusion](https://toolshed.g2.bx.psu.edu/view/iuc/star_fusion): STAR Fusion detects fusion genes in RNA-Seq data after running RNA-STAR STAR-Fusion further processes the output generated by the STAR aligner to map junction reads and spanning reads to a reference annotation set (using a GTF file, ideally the same annotation file used during the STAR genome index building process during the intial STAR setup). https://github.com/STAR-Fusion/STAR-Fusion

* *From [bgruening](https://toolshed.g2.bx.psu.edu/view/bgruening):*
  * [deeptools_plot_enrichment](https://toolshed.g2.bx.psu.edu/view/bgruening/deeptools_plot_enrichment): Wrapper for the deepTools: plotEnrichment. deepTools address the challenge of visualizing the large amounts of data that are now routinely generated from sequencing centers in a meaningful way. For a gallery of images that can be produced and a description of the tools see http://f1000.com/posters/browse/summary/1094053. https://github.com/fidelram/deepTools, doi: 10.1093/nar/gku365,
    * Wikipage: https://github.com/fidelram/deepTools/wiki
    * Repository-Maintainer: Bj&ouml;rn Gr&uuml;ning, https://github.com/fidelram/deepTools
  * [nucleosome_prediction](https://toolshed.g2.bx.psu.edu/view/bgruening/nucleosome_prediction): Prediction of Nucleosomes Positions on the Genome. This tool allows you to submit a genomic sequence and to recieve a prediction of the nucleosomes positions on it, based on the nucleosome-DNA interaction model

* *From [saharlcc](https://toolshed.g2.bx.psu.edu/view/saharlcc):*
  * [isoem2_isode2](https://toolshed.g2.bx.psu.edu/view/saharlcc/isoem2_isode2):  Bootstrapping-Based Estimation of Confidence Intervals of Expression Levels and Differential Expression from RNA-Seq Data

* *From [amadeo](https://toolshed.g2.bx.psu.edu/view/amadeo):*
  * [amadeo](https://toolshed.g2.bx.psu.edu/view/amadeo/amadeo):  This workflow can be applied to microarray data in order to find connections between genetic elements. This workflow extracts genetic information from this microarray data, and finds relationships between cis-regulatory elements located in the promoter under same regulation of gene expression.

* *From [davidvanzessen](https://toolshed.g2.bx.psu.edu/view/davidvanzessen):*
  * [prisca](https://toolshed.g2.bx.psu.edu/view/davidvanzessen/prisca):  PRISCA: PRecISe Clonal Analysis Comparison of clonal sequences in paired samples

# Dependency Definitions

* *From [abretaud](https://toolshed.g2.bx.psu.edu/view/abretaud):*
  * [package_lordec_0_6](https://toolshed.g2.bx.psu.edu/view/abretaud/package_lordec_0_6):  Lordec 0.6 package
