---
autotoc: true
title: Support
---

***Galaxy*** has many options for getting and giving help. 
<br />
<br />
Before submitting an error or posting a question to [Galaxy Biostar](https://biostar.usegalaxy.org/) or one of the [Mailing Lists](/Support#mailing_lists), we ask that you please first review the support resources summarized here and search with our [Custom Tools](/Support#custom_searches) or on tool pages through [Galaxy Biostar](https://biostar.usegalaxy.org/) to see if the same question has come up in the past.
<br />
<br />
If you have help to offer other Galaxy users, *please dive right in* and reply to questions on the [Galaxy Biostar](https://biostar.usegalaxy.org/) forum or [Mailing Lists](/MailingLists), submit tools to the [/Tool Shed](/Tool Shed), add and up vote cards on the [Trello](http://bit.ly/gxytrello) Issue board. Share your expertise by adding your advice and tutorials into our [Wiki](/Wiki Help) or [Shared Data](https://usegalaxy.org/page/list_published) on [usegalaxy.org](http://usegalaxy.org) (Main)! 
<br />
<br />
Galaxy is a [community](/src/Community/index.md) of scientists and developers working together to make great science happen! Communication, feedback, and contributions are *important*.

<div class='left'></div>
<div class='right'>PLACEHOLDER_INCLUDE(/Support/LinkBox)</div>
<div class='right'>
<br />
<a href='http://galaxyproject.org/search/web'><img src='/Images/Logos/GalaxyWebSearch.png' alt='Galaxy Web Search' width="170" /></a> <br />
<br />
<a href='http://galaxyproject.org/search/mailinglists'><img src='/Images/Logos/GalaxyMailingListSearch.png' alt='Galaxy Mailing List Search' width="170" /></a>
<br />
<br />
<a href='http://galaxyproject.org/search/usegalaxy'><img src='/Images/Logos/UseGalaxySearch.png' alt='Using Galaxy Web Search' width="170" /></a>
<br />
<br />
<a href='http://galaxyproject.org/search/getgalaxy'><img src='/Images/Logos/GetGalaxySearch.png' alt='Galaxy administration, tool, and deployment search' width="170" /></a>
</div>
<div class='left'>
# Using Galaxy

## Learning Hub

See our [Learning](/Learn) hub for key coverage of Galaxy user interface concepts, data, and tools. Review "Shared Data &rarr; Published Pages" on the [/Main](/Main) server [usegalaxy.org](http://usegalaxy.org) for publication supplementals and tutorials. Watch the short **[Learn](http://vimeo.com/galaxyproject/learn)** screencast for a learning resource overview.

## Screencasts

[Screencast](/src/Learn/Screencasts/index.md) videos demonstrate the step-by-step for a range of topics. Packed with tips and methods usable across analysis workflows plus presentations and tutorials for administrations, these are a great resource for both the scientific and technical Galaxy communities.

**Watch MOST videos** at *Vimeo* here at **[Galaxy Project on Vimeo](http://vimeo.com/user20484153)**

## Tutorials

[Tutorials](http://main.g2.bx.psu.edu/u/james/p/exercises) embedded inside of Galaxy. Created by Galaxy's scientists, these are packed with example [datasets](/Learn/Managing Datasets), histories, visualizations, and workflows to import and experiment with.

## FAQs
PLACEHOLDER_INCLUDE(/FAQs/LinkBox)

[FAQs](/src/Learn/FAQ/index.md) *are always under active development*. Both scientific and technical users are encouraged to add in their own tips and best practices for new and existing data types, tools, workflows, set-up, and administration.

## Custom Searches
Finally, there are the ***[Galaxy Custom Searches](/News/Custom Galaxy Search)***. The ***[MailingList](http://galaxyproject.org/search/mailinglists)*** search finds all related prior Q & A from the [Galaxy Biostar](#biostar) forum and any [Galaxy Project mailing lists](/src/MailingLists/index.md). The ***[UseGalaxy](http://galaxyproject.org/search/usegalaxy)*** search finds all online resources for information about using Galaxy.  This includes this wiki, tool help and shared Galaxy objects at [UseGalaxy.org](http://usegalaxy.org), and [Mailing Lists](/src/MailingLists/index.md).

## Biostar
Galaxy has teamed up with [Biostar](http://biostars.org) to create a Galaxy User support forum at **[https://biostar.usegalaxy.org](https://biostar.usegalaxy.org)**.
<br />
 **[How to login and use...](/src/Support/Biostar/index.md)**
<div class='right'><a href='https://biostar.usegalaxy.org/'><img src='/Images/Logos/GalaxyBiostar.png' alt='Galaxy Biostar' width="250" /></a></div>
We want to create a space where researchers using Galaxy can come together and share both scientific advice and practical tool help.  Whether on **[http://usegalaxy.org](http://usegalaxy.org)**, a **[Cloudman](http://usegalaxy.org/cloud)** instance, or any **[other](/PublicGalaxyServers) [Galaxy](http://getgalaxy.org)**, if you have something to say about **Using Galaxy**, this is the place to do it!

## Mailing Lists
Galaxy has one public mailing lists for questions, one private mailing list for bug reports, and one announcement mailing list. Please do not post questions through the **[Galaxy Issue Board](/Issues)**; these will only be redirected. Manage subscriptions and learn more about these list at the **[Mailing Lists](/src/MailingLists/index.md)** home page. See also:
<div class='right'>![](/Images/Logos/MailmanLogoSmall.png)</div>
* [Error from tools: Sending in a specific-use bug report](/Support#error_from_tools)
* [Starting a technical tool, local, cloud instance, or development thread](/Support#starting_a_technical_tool2c_local2fcloud_instance2c_or_development_thread)
* [Reporting/vetting a software bug](/Support#reporting_a_software_bug)

**NEW!**: Galaxy scientific user and tool help has moved to **[Galaxy Biostar](/src/Support/Biostar/index.md)**. The galaxy-user mailing list has been retired.

## Galaxy Issue Board

<div class='right'><a href='/Issues'><img src='/Images/Logos/TrelloLogo300.png' alt='Galaxy Issue Board' width="140" /></a></div>

The Galaxy Project uses [Trello](http://bit.ly/gxytrello) for [issue tracking and feature requests](/src/Issues/index.md).
The [Galaxy Issue Board](/Issues) supports [issue creation](http://galaxyproject.org/trello), [commenting](/Issues#add-a-comment), and [voting](/Issues#vote) on issues.

## About Galaxy
<div class='right'><a href='http://galaxyproject.org'><img src='/Images/Screenshots/GalaxyProjectOrg.png' alt='GalaxyProject.org' width="300" /></a></div>
**Galaxy is an open, web-based platform for data intensive biomedical research.**
<br />
<br />
**[Galaxy](/src/FrontPage)**'s goal is to be *accessible*, *reproducible*, and *transparent*. 
* **Accessible:** Users without programming experience can easily specify parameters and run tools and workflows.
* **Reproducible:** Galaxy captures information so that any user can repeat and understand a complete computational analysis.
* **Transparent:** Users share and publish analyses via the web and create Pages, interactive, web-based documents that describe a complete analysis.

There are many [Choices](/BigPicture/Choices) when you want to learn more **[about using Galaxy](/src/Support/AboutGalaxy/index.md)**!

In addition to using the [public Galaxy server](http://usegalaxy.org), you can also install your [own instance of Galaxy](/Admin/GetGalaxy), or create an [instance of Galaxy on the cloud using CloudMan](/CloudMan) and explore all [/Cloud](/Cloud) resources including [AWS in Education](http://aws.amazon.com/grants/) grants.  Another option is to use one of the ever-increasing number of [/Public Galaxy Servers](/Public Galaxy Servers) hosted by other organizations. And if you or your lab is looking to run their own local instance that is "ready-to-go", learn more about the [Slipstream Appliance: Galaxy Edition](http://www.bioteam.net/slipstream/galaxy-edition).

The GALAXY Framework at the highest level is a set of reusable software components. Learn more about [Galaxy's Architecture](/Admin/Internals/Implementation Info).

<div class='left'><a href='http://usegalaxy.org'><img src='/Images/Logos/GalaxyLogoHighRes.png' alt='UseGalaxy.org' width="200" /></a></div>
[Galaxy Project](/GalaxyProject) • [Big Picture](/BigPicture) • [/Community](/Community) • [Get Galaxy](/Admin/GetGalaxy) • [/CloudMan](/CloudMan) • [Tool Shed](/ToolShed) • [/Develop](/Develop) • [News Briefs](/DevNewsBriefs) • [Servers](/PublicGalaxyServers) • [/Learn](/Learn) • [/Support](/Support) • [Galaxy Biostar](http://biostar.usegalaxy.org) • [/News](/News) • [Twitter](/GalaxyOnTwitter) • [/Events](/Events) • [/Teach](/Teach) • [/Issues](/Issues) • [Cite](/CitingGalaxy) • [About Galaxy](/Support/AboutGalaxy) • [Galaxy Team](/src/GalaxyTeam/index.md)
<br />
<br />

The [Galaxy Team](http://wiki.galaxyproject.org/GalaxyTeam) is a part of the [Center for Comparative Genomics and Bioinformatics](http://www.bx.psu.edu/) at [Penn State](http://www.psu.edu/), and the [Department of Biology](http://www.bio.jhu.edu/) at [Johns Hopkinis University](http://www.jhu.edu/). 

[Galaxy](http://usegalaxy.org ) is supported in part by [NSF](http://www.nsf.gov/), [NHGRI](http://www.genome.gov/), the [Huck Institutes of the Life Sciences](http://www.huck.psu.edu/), and [The Institute for CyberScience at Penn State](http://www.ics.psu.edu/), and [Johns Hopkins University](http://www.jhu.edu/).

The [public Main instance](/Main) of Galaxy at [http://usegalaxy.org](http://usegalaxy.org) utilizes infrastructure generously provided by the [iPlant Collaborative](http://www.iplantcollaborative.org/) at the [Texas Advanced Computing Center](https://www.tacc.utexas.edu/), with support from the [National Science Foundation](http://www.nsf.gov/).

<br />
---
<br />

# Common Solutions
<br />
<br />
## Tool help
Galaxy is designed to have a simplified tool interface while still retaining maximum functionality. [Read more...](/src/Support/ToolHelp/index.md)

<br />
<br />

## Getting an account
Having your own account on the public **[/Test](/Test)** and/or **[/Main](/Main)** server means that you can save histories, work with more data, associate an [OpenID](/Learn/OpenID Login), and get the most out of Galaxy's functionality. Be sure to note that the public [/Test](/Test) and [/Main](/Main) instance **usage policies are <div class='red'>one account per user<div class='grey'>**, as stated in our **[Terms and Conditions](https://usegalaxy.org/static/terms.html)**. Also, make sure your email address is valid so that you can confirm your new account (emails are case sensitive) and so that our administrator can contact you if needed (rare, but you'd want the email!). 

Watch the **[Accounts on Main](http://vimeo.com/galaxyproject/accounts)** video for a quick how-to and see our **[User Accounts](/Learn/User Accounts)** wiki for more help.

## Finding a tool
At the top of the left tool panel, type in a tool name or data type into the tool search box. Shorter keywords find more choices. Can't find the tool you want? Try looking in the **[Tool Shed](http://toolshed.g2.bx.psu.edu)**. New tools are added all the time that can be used in **[local](/Admin/GetGalaxy)** or **[cloud](/src/CloudMan/index.md)** Galaxy instances.
<br />
<br />

## Loading data
Data is loaded using the tools in the ***Get Data*** tool group. To load your own local data or data from another source, use the tool ***Get Data &rarr; Upload File*** - watch the **[Get Data: Upload File](http://vimeo.com/galaxyproject/upload)** video to see exactly how it works. Want to practice import/export functions with small sample data? Import the [Upload example history here](https://usegalaxy.org/u/usinggalaxy/h/upload). 
* Each file loaded creates one dataset in the history.
* The maximum size limit is 50G (uncompressed). 
* Most individual file compression formats are supported, but `.tar` archives are not.

***Get Data &rarr; Upload*** methods:
* Load by "browsing" for a local file. Only good for **very small** [datasets](/Learn/Managing Datasets). ( < 2G, but often works best for smaller). If you are having problems with this method, try FTP.
* Load using an **HTTP URL** or **FTP URL**.
* Load a few lines of plain text.
* [Load using FTP](/Learn/Upload via FTP). Either line command or with a desktop client.

### Upload tips
* **Data [quota](/Learn/Managing Datasets#data_size_and_disk_quotas) is at limit**, so *no new data can be loaded*. Disk usage and quotas are reported at ***User &rarr; Preferences*** when logged in. 
* **Password protected data** will require a special URL format. Ask the data source. Double check that it is *publicly accessible*.
* Use ***[FTP](/Learn/Upload via FTP)***, not ***SFTP***. Check with local admin if not sure.
* **No HTML content.** The loading error generated may state this. Remove HTML fields from your dataset before loading into Galaxy or omit HTML fields from the query if importing from a data source (such as Biomart).
* Compression types **.gz/.gzip, .bz/.bzip, .bz2/.bzip2, and single-file .zip are supported.**
* Only the **first file in any compressed archive** will load as a **[dataset](/Learn/Managing Datasets)**.
* Data must be ** < 50G ** (uncompressed) to be successfully uploaded and added as a dataset to a history, from any source.
* **Is the problem the dataset format or the assigned datatype?** Can this be corrected by editing the datatype or converting formats? See [/Learn/Managing Datasets](/Learn/Managing Datasets) for help or watch the screencast above for a how-to example.
* **Problems in the first step working with your loaded data?** It may not have *uploaded* completely. If you used an FTP client, the transfer message will indicate if a load was successful or not and can often restart interrupted loads. This makes FTP a great choice for slower connections, even when loading small files.
<br />

## Downloading data
* **Download datasets** by clicking on the disk icon inside the [dataset](/Learn/Managing Datasets). Good for smaller sizes in all browsers.
* **Download entire histories** by selecting *"Export to File"* from the History menu, and clicking on the link generated.
* **Transfer entire histories** by selecting *"Export to File"* from the History menu, generating the link, coping the link in the "from" Galaxy instance, **then** in the "to" Galaxy instance select "Import from File" from the History menu, and paste in the link into the new form. 
* The video **[Datasets 1](http://vimeo.com/galaxyproject/datasets1)** includes help about different datatypes and what to expect in the download icon (one file or two!).
 
### Download tips
* ***How can I download larger datasets?***

**Browser option:** use [Google Chrome](http://www.williams.edu.au/) and click on the disc icon (this browser supports continuous download.
<br /><br />
**Utility option:** from a shell/unix/terminal window on your computer use **[wget](https://www.gnu.org/software/wget/manual/html_node/Download-Options.html#Download-Options)** or **[curl](http://en.wikipedia.org/wiki/CURL)**.

The link can be obtained by right clicking the floppy disk icon inside a history item and choosing "Copy Link Location" (for most datasets) or "Download Dataset/Download bam_index" (for BAM datasets there are two downloads). Once you have the `<link>`, type this (where "$" indicates the terminal prompt), so that the `<link>` is inside of single quotes. Like many commands, there are many options. These are examples commonly used with Galaxy.

```
  $ wget -O '<link>' 
  $ wget -O --no-check-certificate '<link>' # ignore SSL certificate warnings
  $ wget -c '<link>'                        # continue an interrupted download
```


```
  $ curl -o outfile '<link>' 
  $ curl -o outfile --insecure '<link>'     # ignore SSL certificate warnings
  $ curl -C - -o outfile '<link>'           # continue an interrupted download
```



<br />
<br />
## Dataset status and how jobs execute
The Galaxy user interface (UI) has been designed to communicate job execution status through visual cues and concise meesages. Learn more about how interpret these cues and to use strategies to maximize throughput by [reading more here...](/Support/JobStatus)

When a tool is executed, one or more new **[datasets](/Learn/Managing Datasets)** are added to a history. The same is true when a **workflow** is executed. *If using the public [Main](/src/Main/index.md) Galaxy instance, the most effective strategy when running jobs on the shared resource is to start jobs (or workflows), and then leave them alone to execute until completion.* 
<br />
<br />
When work is urgent during peak-usage times on the public **[/Main](/Main)** Galaxy instance, a **[/CloudMan](/CloudMan)** instance is a quick-to-implement alternative. For large scale and/or urgent ongoing work, a **[/CloudMan](/CloudMan)**, **[Local](/Admin/GetGalaxy)**, or **[SlipStream Galaxy](http://www.bioteam.net/slipstream/galaxy-edition)** each have advantages as a longer-term solution. [Read more ...](/src/BigPicture/Choices/index.md)
<br />
<br />
*So, how does the processing of tool jobs on **[/Main](/src/Main/index.md)** actually work?*
* The **color** of a dataset designates the current status of the underlying job.
  * **green** - the job completed successfully
    * The resulting data is ready to be used in visualizations, available as input to tools, can be downloaded, or utilized for any other downstream purpose.
  * **yellow** - the job is in progress
    * If you are using the public [Main](/Main) Galaxy instance, this job is running on one of our clusters. Different types of tools send jobs to different clusters appropriate for the requirements of each tool. Some tools are more compute intensive than others and significant resources are dedicated to job processing. Jobs have up to 72 hours to complete, if they run longer than this they will fail with a "wall-time" error and turn *red*. Examining tool paramaters is the first option, less sensitive parameters may result in an equally acceptable result, but use less resource. If that is not appropriate or does not succeed, a [CloudMan Galaxy](/CloudMan) or [Local Galaxy](/src/Admin/GetGalaxy/index.md) with sufficient resources may be the solution. 
  * **light blue** - the job is paused
    * This indicates either an input has a problem or that you have [exceeded disk quota](/Main#user_account_quotas) set by the administrator of the Galaxy instance you are working on. 
    * If there is an input problem, correct the problem (often by re-run an upstream job) and click on the tool form option to "resume dependencies". You will not need to stop or restart downstream jobs in most cases (permit paused jobs to start, as inputs datasets become available, through this method).
    * If you need to make room, [permanently delete unneeded data](/Learn/ManagingDatasets#delete_vs_delete_permanently). If you are using the public [Main](/src/Main/index.md) Galaxy instance, disk quotas are defined [here](/src/Main/index.md). You will not need to stop or restart jobs while doing this, unless you are filtering your work to prevent exceeding quota again (only purging, not restarting at this time).
  * **grey** - the job is waiting to run
    * If you are using the public [Main](/src/Main/index.md) Galaxy instance, this job is queued, waiting for an opening on the appropriate cluster. **It is *very important* to allow queued jobs to remain queued, and to not delete/re-run them**. If re-run, this not only moves the new job back to the end of the queue, effectively lengthening the wait time to execute, but if done repeatedly, the volume of "executing deleted" jobs can create additional work processes in the history as these are cleared away, using up resources, and can cause additional delays. 
  * **red** - the job has failed
    * There can be many reasons for this, see the next section, **[Error from tools](/Support#error_from_tools)** for details.
  * **bright blue with moving arrow** - (applies to "Get Data &rarr; Upload File" tool only) - the job is queuing or running
    * The job may run immediately, or may turn *grey* if the server is busy, meaning that **guidelines for grey jobs apply**, and these *grey* datasets should never be deleted/re-run, for the same reasons explained above.
    * An upload job that seems to stay in the "*bright blue with moving arrow*" state for a very long time generally indicates that the file being loaded is too large for the method used (specifically, a browsed-file upload) and [FTP](/Learn/Upload via FTP) should be used instead. This is the only active job that should be deleted under normal usage, as it will never complete (no file over 2G will ever load via file browser upload).

## Shared and Published data
Have you been asked to share a history? Or has someone shared a workflow with you but you're not sure where to find it? Or maybe you just want to find out more about how publishing your work in Galaxy can be used to support your next publication? Watch the how to **[Share and Publish](http://vimeo.com/galaxyproject/sharepublish)** screencast and [read more here](/Learn/Share).

## Error from tools
***[Dataset](/Learn/Managing Datasets) format problems are the #1 reason that tools fail.*** Most likely this problem was introduced during the initial data upload. Double check the [dataset](/Learn/Managing Datasets) against Galaxy's [datatypes](/Learn/Datatypes) or external specifications. In many cases, the format issues can be corrected using a creative combination of Galaxy's tools.

### Troubleshooting tool errors
* Verify the *size/number of lines or md5sum between the source and Galaxy*. Use **Line/Word/Character count of a dataset** or **Secure Hash / Message Digest on a dataset** to do this.
* Look at the *end of your file*. Is it complete? Are there extra empty lines? Use **Select last lines from a dataset** with the default **10** to check.
* Check errors that come from tools such as the **FASTQ Groomer**. Many tools report the *exact problem* with *exact instructions* for corrections. 
* Is the format to specification? Is it recognized by Galaxy? By the target tool or display application? Check against the [Galaxy Datatypes](/Learn/Datatypes) list. 
  * Are you using a **[Custom Reference Genome](/Learn/CustomGenomes)**? Have you tried the quick [Troubleshooting](/Learn/CustomGenomes#troubleshooting) tips on the wiki?
  * Note: not all formats are outlined in detail as they are common types or derived from a particular source. Read the target tool help, ask the tool authors, or even just google for the most current specification.
* Is the problem the [dataset](/Learn/Managing Datasets) format or the assigned datatype? Can this be corrected by editing the datatype or converting formats? Often a combination of tools can correct a formatting problem, if the rest of the file is intact (completely loaded).
* Is the problem a scientific or technical problem? Also see [#interpreting-scientific-results](#interpreting-scientific-results) to decide.
  * *Example* **NGS: Mapping** tools:  On the tool form itself is a short list of help plus links to publications and the tool author's documentation and/or website. If you are having trouble with **Bowtie**, look on this tool's form for more information, including a link to this website: http://bowtie-bio.sourceforge.net/index.shtml.
  * *Example* **NGS: RNA Analysis** tools: See the *[galaxy-rna-seq-analysis-exercise](http://main.g2.bx.psu.edu/u/jeremy/p/galaxy-rna-seq-analysis-exercise)* tutorial and *[transcriptome-analysis-faq](http://main.g2.bx.psu.edu/u/jeremy/p/transcriptome-analysis-faq)*. If these do not address the problem, then contacting the tool authors is the next step at: [tophat.cufflinks@gmail.com](tophat.cufflinks@gmail.com).
  * *Example* **NGS: SAM Tools** tools: SAMTools requires that all input files be to specification ([/Learn/Datatypes](/Learn/Datatypes)) and that the same *exact* reference genome is used for all steps. Double checking format is the first check. Double checking the the same exact version of the reference genome is used is the second check. The last double check is that the number of jobs and size of data on disk is under quota. Problems with this set of tools is rarely caused by other issues.
* Tools for fixing/converting/modifing a [dataset](/Learn/Managing Datasets) will often include the [datatype name](/Learn/Datatypes). Use the tool search to locate candidate tools, likely in tool groups ***Text Manipulation, Convert Formats, or NGS: QC and manipulation***. 
* The most commonly used tools for investigating problems with upload, format and making corrections are:
    **TIP**: use the Tool search in top left panel to find tools by keyword
  * ***Edit Attributes*** form, found by clicking a dataset's ![](/Images/Icons/pencil.png) icon
  * ***Convert Format*** tool group
  * **Select first lines from a dataset**
  * **Select last lines from a dataset**
  * **Line/Word/Character count of a dataset**
  * **Secure Hash / Message Digest on a dataset**
  * **FASTQ Groomer**
  * **FastQC**
  * **Tabular to FASTQ**, **FASTQ to Tabular**
  * **Tabular to FASTA**, **FASTA to Tabular**
  * **FASTA Width formatter**
  * ***Text Manipulation*** tool group
  * ***Filter and Sort*** tool group

## Tool doesn't recognize dataset
Usually a simple datatype assignment incompatibility between the [dataset](/Learn/Managing Datasets) and the tool. Expected input [datatype](/Learn/Datatypes) format is explained on the Tool form itself under the parameter settings. ***Convert Format*** or modify the datatype using the dataset's ![pencil](/Images/Icons/pencil.png) icon to reach the ***Edit Attributes*** form. You may need to first create a **[Custom Build](/Learn/CustomGenomes#custom_builds)** when using a [Custom Reference Genome](/Support#custom_reference_genome).

## Dataset special cases
### FASTQ Datatype QA
* If the required input is a [FASTQ](/Learn/Datatypes#fastq) datatype, and the data is a newly uploaded FASTQ file, run **FastQC** then **FASTQ Groomer** as first steps, then continue with your analysis. Watch the **[FASTQ Prep Illumina](http://vimeo.com/galaxyproject/fastqprep)** screencast for a walk-through.
  * *If you are certain that the quality scores are already scaled to Sanger Phred+33* (the result of an Illumina 1.8+ pipeline), the datatype ".fastqsanger" can be directly assinged. Click the ![pencil](/src/Images/Icons/pencil.png) icon to reach the ***Edit Attributes*** form. In the center panel, click on the "Datatype" tab (3rd), enter the datatype ".fastqsanger", and save. Metadata will assign, then the dataset can be used. 
  * *If you are not sure what type of FASTQ data you have*, see the help directly on the **FASTQ Groomer** tool for information about types.
    * * For Illumina*, first run **FastQC** on a sample of your data. The output report will note the quality score type interpreted by the tool. If not ".fastqsanger", run **FASTQ Groomer** on the entire dataset. If '.fastqsanger", just assign the datatype.
    * *For SOLiD*, run **NGS: Fastq manipulation &rarr; AB-SOLID DATA &rarr; Convert**, to create a ".fastqcssanger" dataset. If you have uploaded a color space fastq sequence with quality scores already scaled to Sanger Phred+33 (".fastqcssanger"), first confirm by running **FastQC** on a sample of the data. Then if you want to double-encode the color space into psuedo-nucleotide space (required by certain tools), see the instructions on the tool form **Fastq Manipulation** for the conversion. 
  * *If your data is [FASTA](/Learn/Datatypes#fasta)*, but you want to use tools that require [FASTQ](/Learn/Datatypes#fastq) input, then using the tool **NGS: QC and manipulation &rarr; Combine FASTA and QUAL**. This tool will create "placeholder" quality scores that fit your data. On the output, click the ![pencil](/src/Images/Icons/pencil.png) icon to reach the ***Edit Attributes*** form. In the center panel, click on the "Datatype" tab (3rd), enter the datatype ".fastqsanger", and save. Metadata will assign, then the dataset can be used. 
### Tabular/Interval/BED Datatype QA
* If the required input is a [Tabluar](/Learn/Datatypes#tabular) datatype, other datatypes that are in a specialized tabular format, such as .bed, .interval, or .txt, can often be directly reassigned to tabular format.  Click the ![pencil](/src/Images/Icons/pencil.png) icon to reach the ***Edit Attributes*** form.  In the center panel, using tabs to navigate, change the datatype (3rd tab) and save, then label columns (1st tab) and save. Metadata will assign, then the dataset can be used.
* If the required input is a [BED](/Learn/Datatypes#bed) or [Interval](/Learn/Datatypes#interval) datatype, the reverse (.tab &rarr; .bed, .tab &rarr; .interval) may be possible using a combination of ***Text Manipulation*** tools, to create a [dataset](/Learn/Managing Datasets) that matches the BED or Interval [datatype](/Learn/Datatypes) specifications.

## Reference genomes
Using the ***same exact reference genome*** for all steps in an analysis is often ***mandatory*** to obtain accurate results. To use the reference genomes available on [usegalaxy.org (Main)](http://usegalaxy.org), get the genome from our **[rsync server](/src/Admin/UseGalaxyRsync/index.md)**.
### Detecting Genome Mismatch Problems
* *How can I tell if I have a **reference genome mismatch problem**?*
  * There isn't one single error that points to this problem. But, if you are running a tool for the first time using a newly uploaded dataset, and an [error occurs](/Support#error_from_tools) or more likely simply [unexpected results](/Support#interpreting_scientific_results) are produced - double checking the reference genome would be a good choice. 
### Correcting Chromosome Identifier Conflicts
* *I suspect there is a problem with the identifiers but how can I check? Or better, how can I __fix the problme__?*
  * A quick way to check for this issue is to compare the chromosome identifiers in the input datasets to each other and to the reference genome used (or intended to be used). 
  * Even small differences in identifiers can cause tools to fail, produce warnings, or create incomplete results. This is the second most common cause of usage-related tool failures ([input format problems](/Support#error_from_tools) are the first).
  * Using an Ensembl-based chromosome identifier file on Galaxy [/Main](/Main) with a locally cashed reference genome? Most built-in, native, reference genomes are sourced from [UCSC](http://genome.ucsc.edu) and have UCSC-based identifier names. When using inputs with both versions of identifiers in the same analysis pipeline, there will almost certainly be errors or unexpected results. But, in many cases, inputs from the history can be adjusted to match the cached data, all within Galaxy. [Read more about how...](/src/Support/ChromIdentifiers/index.md)
  * *Why isn't my Ensembl GTF compatible with Cufflinks and how can I use Ensembl GTFs with Cufflinks?*
    * First, determine if an Ensembl GTF is the best choice. If an iGenomes version is available, this has advantages due to the addition of specific attributes utilized by the RNA-seq Tuxedo pipeline. Check at the *Cufflinks* website [here](http://cufflinks.cbcb.umd.edu/igenomes.html). 
      * Download the .tar file locally, uncompress it, then upload *only* the .gtf file to Galaxy. Loading .tar archives is not supported and has unpredictable outcomes (sometimes the first file in the archive will load - but this is not the file you need, sometimes only a portion of the first file will load - without a warning, and other times an upload error will result: none of these cases should be reported as a bug report/tool error).
      * For certain genomes, the reference annotation GTF file is available on the public [/Main](/Main) Galaxy instance, [http://usegalaxy.org](http://usegalaxy.org), under *Shared Data -> Data Libraries -> iGenomes*. 
    * Next, if you want to proceed, [confirm that your identifiers are a good candidate](/Support/ChromIdentifiers#reference_genome_contents) for the addition of the "chr" adjustment, then use the workflow available in the [Transcriptome Analaysis FAQ](http://usegalaxy.org/u/jeremy/p/transcriptome-analysis-faq#faq5).
### Avoiding Genome Mismatch Issues
* *When **moving between instances**, what can be done to mitigate the risk of using the wrong assembly?*
  * When moving between a **Galaxy [CloudMan AMI](/CloudMan) and the public [/Main](/src/Main/index.md) Galaxy** instance, just make sure the database name is the same. If the assigned database name is the same, the content of the reference genome is the same.
  * When moving between a **[local Galaxy](/Admin/GetGalaxy) and the public [/Main](/src/Main/index.md) Galaxy** instance, there are a few choices:
    * Consider using our version of commonly used genomes, [available through our rsync server](/Admin/Data Integration#get_the_data)
    * [Make your own indexes](/Admin/NGS Local Setup), but check against our [rsync indexes](/Admin/Data Integration#get_the_data) to be aware of differences
    * When there are differences, use your own genome as a [Custom Reference Genome](/Support#custom_reference_genome) with tools
  * When moving between a **[local Galaxy](/Admin/GetGalaxy) and a Galaxy [CloudMan AMI](/src/CloudMan/index.md)**, the same guidelines as immediate above (for [/Main](/Main)) would apply, since [/Main](/Main) and the [CloudMan AMI](/src/CloudMan/index.md) are based off the same content
* ***How do I load a reference genome?***
  * Use FTP - details are [here...](/FTPUpload) and troubleshooting help is [here...](/Support#trouble_loading_data)
  * If your genome is small (bacterial, etc.), using it as a [Custom Reference Genome](/Support#custom_reference_genome) is the quickest way to to get it into Galaxy and to start using it with tools.

## Reporting tool errors
* If running a tool on the [public Galaxy server](/src/Main/index.md) (i.e., http://usegalaxy.org) is resulting in an error (the dataset is <div class='red'>red</span>), and you can't determine the root cause from the error message or input format checks:
  * Re-run the job to eliminate transitory cluster issues.
  * Report the problem using the dataset's ![bug](/src/Images/Icons/bug.png) icon. Do not submit an error for the first failure, but leave it ***undeleted*** in your history for reference.
  * **IMPORTANT**: Get the *quickest resolution* by leaving **all** of the input and output datasets in the analysis thread leading up to the error in your history ***undeleted*** until we have written you back. Use ***Options &rarr; Include Deleted Datasets*** and click dataset links to ***undelete*** to recover error datasets before reporting the problem, if necessary. 
    * *Example:* Error with Cufflinks? Leave the ungroomed + groomed FASTQ, Bowtie/Tophat SAM, optional GTF + custom genome, *and* Cufflinks datasets ***undeleted***.
  * Include in the bug report what checks confirmed that data format was not an issue
  * Anything else you feel is relevant to the error
* We do our best to respond to bug reports as soon as possible.
* Please send all email as ***reply-all*** as we work to resolve the error. The galaxy-bugs address we will be corresponding from is internal to the Galaxy team only and we work together to resolve reported problems.
* If you have resolved the issue, a reply to the bug report to let us know is appreciated.

## Interpreting scientific results
A double check against the tool help and documentation is the first step. If the tool was developed by a 3rd party, they are likely the best experts for detailed questions. Tool forms have links to documentation/authors.

### Tools on the Test server
* Tools on [/Test](/src/Test/index.md) will have little to no support help offered.
* [/Test](/Test) tool errors reported as a bug reports ([#error-from-tools](#error-from-tools)) are considered low priority and may not receive a reply. 
* General feedback & discussion threads (instead of questions requiring a reply from the Galaxy team) are welcomed at the [development mailing list](#public-mailing-list-q--a-discussions). 
* Exceptions are possible. Sometimes community users help to test-drive new functionality. If you are interested in this type of testing for a particular tool, contact us on the [development mailing list](#public-mailing-list-q--a-discussions). 

### Tools on the Main server: RNA-seq
* *Example* &rarr; ***RNA-seq analysis*** **tools.**
  * **Read the Galaxy team's publication, then review the live supplement and try the tutorial**
    * Supplement at **[usegalaxy.org](http://usegalaxy.org)**: **[Interactive RNA-seq analyses by visualization with Trackster](https://usegalaxy.org/u/jeremy/p/interactive-rna-seq-with-trackster)**
    * Goecks, et al. **[[http://www.nature.com/nbt/journal/v30/n11/full/nbt.2404.html|]]** *Nature Biotechnology* 30, 1036–1039 (2012) doi:10.1038/nbt.2404
  * **Review and try the Galaxy team's tutorial exercise**
    * **[Galaxy RNA-seq Analysis Exercise](https://usegalaxy.org/u/jeremy/p/galaxy-rna-seq-analysis-exercise)**
  * **Explore the Galaxy community's RNA-seq learning resources**
    * **[Galaxy community RNA-seq Tutorials](/Learn#other_tutorials)**
  * **Read the publication from the tool authors**
    * Trapnell, et al. **[NGS analyses by visualization with Trackster](http://www.nature.com/nprot/journal/v7/n3/full/nprot.2012.016.html)** *Nature Protocols* 7, 562–578 (2012) doi:10.1038/nprot.2012.016
  * **Verify your inputs!** - format or data mismatch issues are the most common problems and are usually easily solved
    * [GTF](/Learn/Datatypes#gtf) dataset formats can widely vary in the 9th **attribute** column. Cufflinks and the other tools in this set expect a certain format for full functionality. [GFF3](/Learn/Datatypes#gff3) datasets have a specific structure, including a unique "ID" attribute. The details are explained in the [file format specifications](/Learn/Datatypes) and at the tool's websites: http://cufflinks.cbcb.umd.edu/faq.html, http://cufflinks.cbcb.umd.edu/ & http://tophat.cbcb.umd.edu/.
    * Using the same reference genome for all steps is very important. Even small differences in chromosome/scaffold names can result in errors. Double check that the naming between the reference genome and any other inputs such as SAM/BAM and GTF datasets all use the same naming conventions. See our [FAQ](http://main.g2.bx.psu.edu/u/jeremy/p/transcriptome-analysis-faq) for more help if this is suspected to be the root cause of an error.
    * Confirming data sources using **[gffread](http://cufflinks.cbcb.umd.edu/gff.html)** locally, before loading data into **Galaxy**, can be one way to discovering where problems are.
  * **Still stuck?** 
    * If the tool form help, publications & supplemental/tutorials including those from the community under [/Learn](/Learn), the [Transcriptome FAQ](http://main.g2.bx.psu.edu/u/jeremy/p/transcriptome-analysis-faq), or tool author's web sites do not address the question or problem, then contacting the tool authors is often the next step for detailed algorithm questions: [tophat.cufflinks@gmail.com](tophat.cufflinks@gmail.com). 
    * You can also try searching the [Galaxy Mailing Lists and Galaxy Biostar](http://galaxyproject.org/search/mailinglists) for prior Q&A, then ask a question at [galaxy Biostar](/Support#starting_a_scientific2c_data2c_or_tool_usage_thread) if the problem is new or you just need Galaxy-specific help. 
    * If you ended up with a failed dataset (red), it is sometimes better to submit that instead as a [tool error (bug report)](/Support#error_from_tools) unless the question is general.   
    * *We can help or guide you to help*. Whenever [sharing](/Learn/Share) or submitting a history for feedback, please be sure to leave the datasets in the analysis thread *undeleted* so that we can offer the best advice.
<br />

## Custom reference genome
Often the quickest way to get your analysis going is to load a custom genome for your own use. Simply upload the [FASTA](/Learn/Datatypes#fasta) file [using FTP](/Learn/Upload via FTP) and use it as the "reference genome from the history" (wording can vary slightly between tools, but most have this option). [ Read more about how to set up and use a Custom Genome](/src/Learn/CustomGenomes/index.md) including how to create a *Custom Build*.<br />
Video: **[Custom Genomes](http://vimeo.com/galaxyproject/customgenome)**
<br />

### Best Practices
* Make sure the reference genome is in [FASTA](/Learn/Datatypes#fasta) format and is completely loaded (see [Trouble loading data](/Support#trouble_loading_data) above).
* Use the same custom genome for all the steps in your analysis that require a reference genome. Don't switch or the data can become mismatched in your files, preventing downstream work.
* To add a custom **Genome Build** so that it can be assigned as a "database" attribute, or to make it known/available to certain tools, create it under "User &rarr; Custom Builds". More details [here...](/Learn/CustomGenomes#custom_builds).
* **TIP**: To modify a [dataset](/Learn/Managing Datasets) to have an unassigned reference genome, use the ![pencil](/src/Images/Icons/pencil.png) icon to "Edit Attributes". On the form, for the attribute **Database/Build:**, set the genome to be " **`unspecified (?)`** ", and submit. Any prior assignments will be removed.
* If you genome is available on [usegalaxy.org (Main)](http://usegalaxy.org), but just not indexed for the tool you want to use, you can get the genome from our **[rsync server](/src/Admin/UseGalaxyRsync/index.md)**. This will ensure that all of your work uses the *same exact reference genome* for all steps in an analysis, a critical part of a successful experiment. 
* If you find that there are in downstream [tool errors](/Support#error_from_tools) after using a *Custom reference genome* in an upstream tool on [usegalaxy.org (Main)](http://usegalaxy.org), this is good cause to suspect that there is a **[ reference genome mismatch problem](/Support#reference_genomes)**. This generally means that the *Custom* genome needs to be changed to use ours, or that you need to use the *Custom* genome for all downstream tools, too.

### Quick genome access
* If your genome is small (bacterial, etc.), using it as a [Custom Reference Genome](/src/Learn/CustomGenomes/index.md) is the quickest way to to get it into Galaxy and to start using it with tools.
* Obtain a [FASTA](/Learn/Datatypes#fasta) version, load using [using FTP](/FTPUpload), and use from your history with tools.

### Tools on the Main server: Extract DNA
* *Example* &rarr; ***Fetch Sequences: Extract Genomic DNA***
  * Start by loading the custom reference genome in [FASTA](/Learn/Datatypes#fasta) format into your history as a dataset, [using FTP](/Learn/Upload via FTP) if the dataset is over **2G** in size.
  * Load or create an appropriate [Interval](/Learn/Datatypes#interval), [BED](/Learn/Datatypes#bed), or [GFF](/Learn/Datatypes#gff) coordinate dataset into the same history. 
  * On the **Extract Genomic DNA** tool form, you will use the options: 
    * "Source for Genomic Data:" as  "History"
    * next,  for the new menu option "Using reference file", select the fasta dataset of your target genome from your active history
<br />
---
<br />


# Public mailing list Q & A discussions
<br />

<div class='right'><br /><a href='/MailingLists#searching'><img src='/Images/Logos/GalaxyMailingListSearch.png' alt='Search all' width="175" /></a></div>
## Searching prior Q & A
Still need help not covered by the tool help, the [Learning Hub](/Learn), a [Screencast](/Learn/Screencasts), a [Tutorial](http://main.g2.bx.psu.edu/u/james/p/exercises), or an [FAQ](/src/Learn/FAQ/index.md)? 
* Start with a search in our [mailing list archives](/MailingLists#searching) to see if this question has come up before.
* If you have a development topic to discuss, your data/tool situation has not come up before, and/or troubleshooting has failed (including at least one re-run, as explained in [Error from tools](/Support#error_from_tools) above), post to a list or [Galaxy Biostar](/src/Support/BioStar/index.md)
<br />

***Note:*** **If your question is about an error on [/Main](/Main) for a job failure**, start by reviewing the **troubleshooting help for [Tool Errors](#error_from_tools)**. If data input and the job error message don't resolve the issue, please use the **[tool error](#reporting_tool_errors) submission form** from the <div class='red'>red</span> error dataset, instead of starting a public mailing list discussion thread (**do not delete error datasets**). [Read more ...](#error_from_tools)
<br />
<br />

## What to include in a question
1. Where you are using Galaxy: [/Main](/src/Main/index.md), other public, local, or cloud instance
1. End-user questions from [/Test](/src/Test/index.md) are generally not sent/supported - *Test is for breaking*
1. If a local or cloud instance, the distribution or galaxy-central hg pull #
1. If on [/Main](/src/Main/index.md), date/time the initial and ru-run jobs were executed
1. If there is an example/issue, exact steps to reproduce
1. What troubleshooting steps (if a problem is being reported) you have tested out
1. If on [/Main](/src/Main/index.md), you may be asked for a shared history link. Use ***Options &rarr; Share or Publish***, generate the link, and email it directly back off-list. Note the dataset #'s you have questions about.
1. **IMPORTANT**: Get the *quickest answer* for data questions by leaving **all** of the input and output datasets in the analysis thread in your shared history ***undeleted*** until we have written you back. Use ***Options &rarr; Show Deleted Datasets*** and click dataset links to ***undelete*** to recover datasets if necessary
1. **Always ***reply-all*** unless sharing a private link**
<br />
<br />

## Starting a scientific, data, or tool usage thread
* Do not use a mailing list. Use [Galaxy Biostar](/Support#biostar)

## Starting a technical tool, local/cloud instance, or development thread
* Gather information "What to include in a question" above
* Send an email to [galaxy-dev@bx.psu.edu](galaxy-dev@bx.psu.edu)
* Subscribing to the [Galaxy Development List](http://lists.bx.psu.edu/listinfo/galaxy-dev) is recommended for tool developers and instance administrators
* Discussion threads are open to the entire community and the Galaxy team to answer
* **Always ***reply-all*** unless sharing a private link**
<br />
<br />

<br />
---
<br />

# Reporting a software bug
<br />
If you think you've seen a bug (not an ["Error from tools"](/Support#error_from_tools) ), please **report it** to the [Galaxy Development List](http://lists.bx.psu.edu/listinfo/galaxy-dev).
<br />

Please do not report a new usage problem through the **[Galaxy Issue Board](/Issues)** unless you are fairly certain the problem is software and that it can't be remedied quickly. Sending question to the [galaxy-dev@bx.psu.edu](http://lists.bx.psu.edu/listinfo/galaxy-dev) mailing list for review is most often the best first pass if there is a problem. The great part about this method that if the problem is usage, we can help solve the problem, if the problem is that you are looking for a wrapper or help with a wrapper, the community can offer quick feedback, and if there really is a serious issue with Galaxy itself - it might apply globally, and we often can just fix it right away.

<br />

Bug or [Error from tools](/Support#error_from_tools)? Sometimes it is hard to tell. If you are on the public [/Main](/Main) instance, and ran a tool that produced a <div class='red'>red</span> error dataset, then you will probably want to start by reporting this as a [Tool Error](/Support#reporting_tool_errors), but add in comments about your suspicious about a bug if there is something odd about the job failure.
<br />

## What to include in a bug report
1. Where you are using Galaxy ([/Main](/src/Main/index.md), local, or cloud instance). 
1. Bug reports from [/Test](/src/Test/index.md) are generally not sent
1. If a local or cloud instance, the distribution or galaxy-central hg pull #
1. The date/time the bug was detected
1. Exact steps to reproduce the issue
1. What troubleshooting steps (if any) you have tested out
1. If you can reproduce on [/Main](/src/Main/index.md), you may be asked to send in a tool error report or share a history link. Use ***Options &rarr; Share or Publish***, generate the link, and email it directly back off-list. Note the problem dataset #'s. 
1. **IMPORTANT**: If data is involved, leave **all** of the related datasets in the analysis thread leading up to the bug in your history ***undeleted*** until we have written you back. Use ***Options &rarr; Show Deleted Datasets*** and click dataset links to ***undelete*** to recover error datasets before reporting a bug if necessary. 
1. **Always ***reply-all*** unless sharing a private link**

<br />
---
<br />
</div>
