---
title: [UC Riverside NGS Tutorial](http://manuals.bioinformatics.ucr.edu/workshops/dec-12-16-2013)
---
<div class='center'>
[[<a href='http://manuals.bioinformatics.ucr.edu/workshops/dec-12-16-2013'><img src='/UCR_IIGB_Logo.png' alt='UC Riverside Institute for Integrative Genome Biology'  /></a>
</div>





<div class='deploymentbox'>
 Resource:: **[UC Riverside NGS Tutorial](.)**
 Types:: Tutorial, Datasets
 Domains:: **Variant Calling, RNA-Seq, Intro** 
 Owners:: Neerja Katiyar,  [Tyler Backman](http://facility.bioinformatics.ucr.edu/people/tyler-backman), [Rebecca Sun](http://facility.bioinformatics.ucr.edu/people/rebecca-sun), and [Thomas Girke](http://girke.bioinformatics.ucr.edu/), [UC Riverside](http://labs.bioinformatics.ucr.edu/)
 Formats:: Slides, Web pages, Documents
 Date:: 2013/12
</div>

[This tutorial](http://manuals.bioinformatics.ucr.edu/workshops/dec-12-16-2013) introduces Galaxy and then walks through a variant calling example and an RNA-Seq example.


## Needed Tools

Makes use of several "standard" tools.  The variant calling example uses

* BWA, Mpileup, bcftools view

The RNA-Seq example uses

* !TopHat, !CuffDiff

The slides end with a demo of IGV.


## Datasets

Datasets are linked to from the examples.  Datasets are large enough that you may want to preload them in to a data library or shared history on your training server(s).

## Links

* **[Web-based Analysis of Next Generation Sequence Data](http://manuals.bioinformatics.ucr.edu/workshops/dec-12-16-2013)**
  * [Part of Next Generation Data Analysis Marathon](http://manuals.bioinformatics.ucr.edu/workshops/dec-12-16-2013)<br />Includes 
    * [slides](http://biocluster.ucr.edu/~nkatiyar/Galaxy_workshop/Slides/Galaxy_workshop_2013.pdf)
    * [exercises](http://biocluster.ucr.edu/~nkatiyar/Galaxy_workshop/Exercises/Galaxy_workshop_exercises_2013.pdf)
    * [manual](http://manuals.bioinformatics.ucr.edu/home/gui-ngs-analysis)

CategoryTrainingResource
