---
title: Learn Galaxy
---
There are many approaches to learning how to *use* Galaxy. The most popular is probably to just dive in and use it. Galaxy is simple enough to use that you can do many analyses just by exploring the interface. However, you may miss much of the power this way. 

<div class="alert alert-success" role="alert">Have you created or know of a resource that is useful for teaching with Galaxy? Then please share it! This will help others and also help get the word out about your resource. Use [this Google form](http://bit.ly/gxytrainresourceform) to describe your resource. **Also**: consider joining Galaxy Training Network and contributing your tutorial as described [here](https://github.com/galaxyproject/training-material/blob/master/CONTRIBUTING.md)!
</div>

## Tutorials using Galaxy Main

These tutorials are using Galaxy's main site at http://usegalaxy.org:

 - [Galaxy 101](/tutorials/g101) - the basic introduction to Galaxy's interface, its functionality, and workflows. **Start here** if you never used Galaxy before.
 - [Uploading data](/tutorials/upload) - how to get data into Galaxy.
 - [Histories](/tutorials/histories) - in Galaxy uploaded data and analysis results reside within the history pane. This tutorial explains how history works. 
 - [Dataset collections](/tutorials/collections) - modern studies usually include many samples. Collection are designed to simplify complex, multi-sample analyses as shown in this tutorial.
 - [NGS logistics](/tutorials/ngs) - this is an introduction to Galaxy's functionality for the analysis of Next Generation Sequencing data.
 - Variant discovery: [diploid case](/tutorials/var_dip) - this tutorial explains calling sequence variants in **diploid** systems like ... humans.
 - Variant discovery: [haploid case](/tutorials/var_hap) - this tutorial explains calling sequence variants in **haploid** systems like ... bacteria and viruses. 
 - RNA-seq: [Introduction](/tutorials/rb_rnaseq) - an introductory tutorial for transcriptome analysis.
 - RNA-seq: [Discovering and quantifying new transcripts](/tutorials/nt_rnaseq) - an in-depth transcriptome analysis example.
 - ChIP-seq: [A simple ChIP-seq experiment with two replicates](/tutorials/chip) - an example analysis for finding transcription factor binding sites.

## Interactive Tours

Tours guide you through interface elements interactively:

 - [Using Galaxy Interface](https://usegalaxy.org/tours/core.galaxy_ui)
 - [Understanding histories](https://usegalaxy.org/tours/core.history)
 - [Scratchbook introduction](https://usegalaxy.org/tours/core.scratchbook)

## Tutorials by Galaxy Training Network

Thanks to a large [group of wonderful contributors](https://github.com/galaxyproject/training-material/graphs/contributors) there is a constantly growing [set of tutorials](http://galaxyproject.github.io/training-material/) maintained by the [Galaxy Training Network](/src/teach/gtn). These include:

 - [Galaxy Introduction](http://galaxyproject.github.io/training-material/topics/introduction)
 - [Analysis of sequences](http://galaxyproject.github.io/training-material/topics/sequence-analysis/)
 - [Assembly](http://galaxyproject.github.io/training-material/topics/assembly/)
 - [ChIP-seq analysis](http://galaxyproject.github.io/training-material/topics/chip-seq/)
 - [Transcriptomics](http://galaxyproject.github.io/training-material/topics/transcriptomics/)
 - [Variant Analysis](http://galaxyproject.github.io/training-material/topics/variant-analysis/)
 - [Proteomics](http://galaxyproject.github.io/training-material/topics/proteomics/)
 - [Metagenomics](http://galaxyproject.github.io/training-material/topics/metagenomics/)
 - PlantTribes: [Perform comparative and evolutionary analyses of gene families and transcriptomes](/tutorials/pt_gfam)

## Tutorials from Lewis-Sigler Institute @ Princeton 

These tutorials are produced by Lance Parsons at LSI

 - [Visualization with Galaxy and IGV Tutorial](http://bit.ly/10TFBcy)
 - [RNA-Seq Data Analysis Workshop](http://bit.ly/11Qvnwh)
 - [SNP & Indel Detection Tutorial](http://bit.ly/10mulcT)
 - [ChIP-Seq Tutorial](http://bit.ly/13fLyT1)

## Tutorials from around the Web

In addition the following tutorials are available from other contributors:

| Topic |  Authors  |  Posted / Presented  | 
| ----- | -------- | ------------------- | 
| **[Web-based Analysis of Next Generation Sequence Data](http://manuals.bioinformatics.ucr.edu/workshops/dec-12-16-2013)**<div class='indent'>[Part of Next Generation Data Analysis Marathon](http://manuals.bioinformatics.ucr.edu/workshops/dec-12-16-2013)<br />Includes [slides](http://biocluster.ucr.edu/~nkatiyar/Galaxy_workshop/Slides/Galaxy_workshop_2013.pdf), [exercises](http://biocluster.ucr.edu/~nkatiyar/Galaxy_workshop/Exercises/Galaxy_workshop_exercises_2013.pdf), [manual](http://manuals.bioinformatics.ucr.edu/home/gui-ngs-analysis)</div> |  Neerja Katiyar,  [Tyler Backman](http://facility.bioinformatics.ucr.edu/people/tyler-backman), [Rebecca Sun](http://facility.bioinformatics.ucr.edu/people/rebecca-sun), and [Thomas Girke](http://girke.bioinformatics.ucr.edu/), [UC Riverside](http://labs.bioinformatics.ucr.edu/)  |  2013/12  | 
| **[Intro to Using Galaxy for Bioinformatics](https://scholarworks.iu.edu/dspace/bitstream/handle/2022/17204/GalaxyWorkshop091713_t.pdf?sequence=1)** <div class='indent'>Includes a transcriptome assembly example.</div> |  Ganote, Carrie L.; Doak, Thomas  |  2013//09  | 
| **[UC Davis 2013 Bioinformatics Short Course](http://training.bioinformatics.ucdavis.edu/docs/2013/09/short-course-2013/)**<div class='indent'> Includes slides and exercises on *almost everything*. |  [Joe Fass](mailto:jnfass AT ucdavis DOT edu), [Nikhil Joshi](mailto:najoshi AT ucdavis DOT edu), [Jessie Li](mailto:jjsli AT ucdavis DOT edu), [Monica Britton](mailto:mtbritton AT ucdavis DOT edu), [Blythe Durbin-Johnson](mailto:bpdurbin AT phs DOT ucdavis DOT edu)  |  2013/09  | 
| **[Informatics on High Throughput Sequencing Data](http://bioinformatics.ca/workshops/2013/informatics-high-throughput-sequencing-data)**<div class='indent'>Module 5 - Bringing it all together: Galaxy</div> |  [Francis Oullette](http://bioinformatics.ca/person/cbw-experts/francis-ouellette)  |  2013/06  | 
| **[NGS in Galaxy](http://scriptogr.am/ohofmann)**<div class='indent'>Includes [ChIP-Seq](http://scriptogr.am/ohofmann/chip-seq), [RNA-Seq](http://scriptogr.am/ohofmann/rna-seq) and [Exome-Seq](http://scriptogr.am/ohofmann/exome-seq) examples. |  Shannan Ho Sui, Oliver Hofmann, Winston Hide, [Center for Health Bioinformatics](http://compbio.sph.harvard.edu/chb/) at the [Harvard School of Public Health](http://www.hsph.harvard.edu/)  |  2013/05  | 
| **[Introduction to Galaxy 2013 Bootcamp](http://training.bioinformatics.ucdavis.edu/docs/2013/02/bootcamp/galaxy/index.html)** |  Nikhil Joshi, Bioinformatics Core, UC Davis Genome Center  |  2013/02  | 
| **[RNA-seq Analysis in Galaxy](http://www.rna-seqblog.com/events/short-training-classes-rna-seq-analysis-in-galaxy/)** |  [BaRC, Whitehead Institute](http://jura.wi.mit.edu/bio/education/hot_topics/)  |  2013/01  | 
| **[Performing de novo assemblies using the NBIC Galaxy instance](http://www.nbic.nl/uploads/media/de_novo_assemblies_on_NBIC_Galaxy_-_9_jan_2013.pdf)** <div class='indent'> Part of [de novo Assembly from NGS Data Course at NBIC](http://www.nbic.nl/education/nbic-phd-school/course-schedule/ngsdenovo/)</div> |  Jan van Haarst (WUR)  |  2013/01  | 
| **[IIHG Bioinformatics Short Course](https://wiki.uiowa.edu/display/galaxyusers/IIHG+Bioinformatics+Short+Course+II)** <div class='indent'>Downloadable PDFs cover *Galaxy Intro and File formats*, *NGS Intro*, *Galaxy Basics*, and *Reproducibility and Collaboration within Galaxy*</div> |  Ann Black-Ziegelbein, Tom Bair, Srinivas Maddhi  |  2013/01  | 
| **[NGS Analysis with Galaxy and IGV](http://manuals.bioinformatics.ucr.edu/home/gui-ngs-analysis)** |  [Tyler Backman](http://facility.bioinformatics.ucr.edu/people/tyler-backman), [Rebecca Sun](http://facility.bioinformatics.ucr.edu/people/rebecca-sun) and [Thomas Girke](http://girke.bioinformatics.ucr.edu/), [UC Riverside](http://labs.bioinformatics.ucr.edu/)  |  2012-12-10  | 
| **[Galaxy Tutorials](https://genome.edu.au/wiki/Learn)**<div class='indent'>Including *RNA-Seq, Variant Detection,* and *Genome Assembly*</div> |  The [Genomics Virtual Lab](https://genome.edu.au/)  |  2012/09-12  | 
| **[Analysis of ChIP-seq data in Galaxy](http://jura.wi.mit.edu/bio/education/hot_topics/galaxy/GalaxyNov2012_ChIP-seq_toPost.pdf)** |  [BaRC, Whitehead Institute](http://jura.wi.mit.edu/bio/education/hot_topics/)  |  2012/11  | 
| **[Next Generation Sequencing Data Analysis](http://nihlibrary.ors.nih.gov/bioinfo/)** (Course no 11) <div class='indent'>"Massively parallel sequencing, also known as next generation sequencing, is a technology enabling high-throughput sequencing of genomes or loci of interest.  This course focuses on a single locus.  It examines the quality of the sequence reads; mapping of reads; and the quality of the mapping.  It also examines sequence variation." ([slides](https://depot.galaxyproject.org/hub/attachments/documents/presentations/2012NIHLibYoungNGSDataAnalysis.pdf)) |  [Lynn Young](http://nihlibrary.nih.gov/Services/Bioinformatics/Pages/biostaff.aspx)  |  2012/09  | 
| **[Automated and reproducible analysis of NGS data (ARANGS12)](https://github.com/rvosa/ngs-workflows/tree/master/doc/slides/day4)**<div class='indent'>Day 4</div> |  [Rutger Vos](http://rutgervos.blogspot.com/), Darin London  |  2012/09  | 
| **[Galaxy, a web based platform for bioinformatics analysis](http://jura.wi.mit.edu/bio/education/hot_topics/galaxy/Galaxy_June2012.pdf)** |  [Bioinformatics & Research Computing @ MIT](http://jura.wi.mit.edu/bio/), as part of their [Hot Topics series](http://jura.wi.mit.edu/bio/education/hot_topics/)  |  2012/06  | 
| **[Using Galaxy for NGS Analysis](http://chagall.med.cornell.edu/galaxy/)** |  [Luce Skrabanek](mailto:las2017 AT med DOT cornell DOT edu)  |  2012/06  | 
| **[RNA-Seq Course v1.0 documentation](http://training.bioinformatics.ucdavis.edu/docs/2012/05/RNA/index.html)** |  [UC Davis Bioinformatics Professional Training Program](http://training.bioinformatics.ucdavis.edu/)  |  2012/05  | 
| **[Analyzing epigenome data in context of genome evolution and human diseases](http://www.springerlink.com/content/l2184206p7583v75/)** |  Feuerbach, *et al.*  |  2012/02  | 
| **[UAB Galaxy RNA Seq Step by Step Tutorial](https://docs.uabgrid.uab.edu/wiki/UAB_Galaxy_RNA_Seq_Step_by_Step_Tutorial)** |  [Curtis Hendrickson](http://www.ccts.uab.edu/pages/bmi/bmi_main.htm)  |  2011-09-16  |
